cms/djangoapps/contentstore/docs/decisions/0003-hybrid-approach-for-public-apis.rst

```diff
@@ -0,0 +1,67 @@
+0003: Hybrid approach for public course authoring APIs
+======================================================
+
+Context
+-------
+
+We are planning to offer public APIs via oauth that allow course authors to create
+and update course content via an external application. The plan for this is outlined in this
+`Spec Demo`_.
+
+.. _`Spec Demo`: https://openedx.atlassian.net/wiki/spaces/COMM/pages/3696066564/Spec+Memo+API-Based+Management+of+edX+Course+Blocks+Outlines+and+Settings+MVP.
+
+Rather than restricting a course to only ever be editable either via API or manually,
+we will offer a hybrid solution where both is possible.
+
+A hybrid solution faces two challenges:
+
+- "out-of-date write conflicts": how do we handle a situation where an API user updates course content,
+  yet there is an implicit conflict because a manual user edited the course recently and now the API user's local files are outdated?
```

Comment:
... user attempts to update a course using a stale course snapshot as a starting point.